File: Tutorials/C#/Tutorials/Tutorial12.cs
using System;
using System.Windows.Forms;
using TutorialsBase;
using NetOffice;
using Excel = NetOffice.ExcelApi;

namespace TutorialsCS4
{
    public partial class Tutorial12 : UserControl, ITutorial
    {
        public Tutorial12()
        {
            InitializeComponent();
        }

        public void Run()
        {

        }

        public void Connect(IHost hostApplication)
        {
            HostApplication = hostApplication;
            propertyGrid1.SelectedObject = NetOffice.Settings.Default;
        }

        public void Disconnect()
        {

        }

        public string Uri
        {
            get { return Program.DocumentationBase + "Tutorial12_EN_CS.html"; }
        }

        public string Caption
        {
            get { return "Tutorial12"; }
        }


        public string Description
        {
            get { return "NetOffice Settings"; }
        }

        public UserControl Panel
        {
            get { return this; }
        }

        internal IHost HostApplication { get; private set; }
    }
}

File: Tutorials/VB/Tutorials/Tutorial12.vb
Imports NetOffice

Public Class Tutorial12
    Implements ITutorial

    Dim _hostApplication As IHost

    Public Sub Run() Implements TutorialsBase.ITutorial.Run

    End Sub

    Public Sub Connect(ByVal hostApplication As TutorialsBase.IHost) Implements TutorialsBase.ITutorial.Connect

        _hostApplication = hostApplication
        propertyGrid1.SelectedObject = NetOffice.Settings.Default
    End Sub

    Public ReadOnly Property Caption As String Implements TutorialsBase.ITutorial.Caption
        Get
            Return "Tutorial12"
        End Get
    End Property

    Public ReadOnly Property Description As String Implements TutorialsBase.ITutorial.Description
        Get
            Return "NetOffice Settings"
        End Get
    End Property

    Public Sub Disconnect() Implements TutorialsBase.ITutorial.Disconnect

    End Sub

    Public ReadOnly Property Panel As System.Windows.Forms.UserControl Implements TutorialsBase.ITutorial.Panel
        Get
            Return Me
        End Get
    End Property

    Public ReadOnly Property Uri As String Implements TutorialsBase.ITutorial.Uri
        Get
            Return FormMain.DocumentationBase & "Tutorial12_EN_VB.html"
        End Get
    End Property

End Class
